src/helpers/darken.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import parseColorString from '../internalHelpers/_parseColorString'
+import rgbToHsl from '../internalHelpers/_rgbToHsl'
+import hslToHex from '../internalHelpers/_hslToHex'
+import hslToRgb from '../internalHelpers/_hslToRgb'
+
+/**
+ * Returns a string value for the darkened color.
+ *
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   background: darken(0.2, '#FFCD64'),
+ *   background: darken(0.2, 'rgba(255,205,100,0.7)'),
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   background: ${darken(0.2, '#FFCD64')};
+ *   background: ${darken(0.2, 'rgba(255,205,100,0.7)')};
+ * `
+ *
+ * // CSS in JS Output
+ *
+ * element {
+ *   background: "#ffbd31";
+ *   background: "rgba(255,189,49,0.7)";
+ * }
+ */
+function darken(amount: number, color: string): string {
+  // parse color string to hsl
+  const hslColor = rgbToHsl(parseColorString(color))
+  const value = {
+    ...hslColor,
+    lightness: Math.max(0, Math.min(1, hslColor.lightness - amount)),
```

Comment:
`Math.max(Math.min))` stuff could be extracted into a common `guard(0, 1, val)` helper, do we need it anywhere else?